File: src/main/java/com/airwaves/airwavesweb/api/TestController.java
package com.airwaves.airwavesweb.api;

import com.airwaves.airwavesweb.datastore.Cluster;
import com.airwaves.airwavesweb.datastore.User;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;
import java.util.Map;
import java.util.stream.Collectors;

@RestController
public class TestController {

    @GetMapping(value = "/testclusters")
    public List<Map<String, Object>> testclusters() {
        return Cluster.getAll().stream().map(x -> x.snapshot().getData()).collect(Collectors.toList());
    }

    @GetMapping(value = "/testusers")
    public List<Map<String, Object>> testusers() {
        return User.getAll().stream().map(x -> x.snapshot().getData()).collect(Collectors.toList());
    }

    @GetMapping(value = "/testreads")
    public Map<String, Object> testreads() {
        return Map.of("User reads", User.reads, "User writes", User.writes, "Cluster reads", Cluster.reads, "Cluster writes", Cluster.writes);
    }
}

File: src/main/kotlin/com/airwaves/airwavesweb/api/TestController.kt
package com.airwaves.airwavesweb.api

import com.airwaves.airwavesweb.datastore.Cluster
import com.airwaves.airwavesweb.datastore.Document
import com.airwaves.airwavesweb.datastore.User
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RestController

@RestController
class TestController {

    @GetMapping("/testclusters")
    fun testClusters(): List<Map<String, Any>> = Cluster.all.map { x -> x.data }

    @GetMapping("/testusers")
    fun testUsers(): List<Map<String, Any>> = User.all.map { x -> x.data }

    @GetMapping("/testreads")
    fun testReads(): Map<String, Any> = mapOf(
            "User reads" to User.reads, "User writes" to User.writes,
            "Cluster reads" to Cluster.reads, "Cluster writes" to Cluster.writes,
            "Document reads" to Document.reads, "Document writes" to Document.writes
    )

    @GetMapping("/clear")
    fun clear() {
        Cluster.all.forEach(Cluster::delete)
        User.all.forEach(User::delete)
    }

}